src/System.Security.Cryptography.Cng/src/Internal/Cryptography/Helpers.cs

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Globalization;
+using System.Security.Cryptography;
+using System.Runtime.InteropServices;
+
+using Microsoft.Win32.SafeHandles;
+
+using ErrorCode = Interop.NCrypt.ErrorCode;
+
+namespace Internal.Cryptography
+{
+    internal static class Helpers
+    {
+        public static byte[] CloneByteArray(this byte[] src)
+        {
+            return (byte[])(src.Clone());
+        }
+
+        //
+        // The C# construct
+        //   
+        //    fixed (byte* p = new byte[0])
```

Comment:
@stephentoub it is explicitly undefined :smile:.  From section 18.6 in C# 3.5 spec: 

> The behavior of the fixed statement is implementation-defined if the array expression is null or if the array has zero elements

The IL though seems to dynamically check the length of arrays and use `null` in case of `0` or a `null` array.